src/Host/Client/Impl/Definitions/IRSessionProvider.cs

```diff
@@ -5,19 +5,28 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.R.Host.Client.Host;
 
 namespace Microsoft.R.Host.Client {
     public interface IRSessionProvider : IDisposable {
-        IRSession GetOrCreate(Guid guid);
+        IRSession GetOrCreate(Guid guid, IRHostBrokerConnector brokerConnector);
```

Comment:
Oh boy. OK. I guess we then may consider at some point reversing composition. From most callers perspective session provider belongs to a particular REPL (usually the active one) or set of sessions that user is currently operating upon. That makes broker primary and session provider its property rather than the other way around. Then caller simply talks to the broker about getting sessions.